### PR TITLE
Feature/32067 month view code cleanup

### PR DESCRIPTION
**Pull request by jazbek**
See:
https://central.tri.be/issues/32067
https://central.tri.be/issues/37143

### CODE REVIEW

File: src/Tribe/Query.php [hunk 1]
```diff
@@ -712,8 +704,8 @@ public static function getHideFromUpcomingEvents() {
 		 * @return array The counts array.
 		 */
 		public static function getEventCounts( $args = array() ) {
+			_deprecated_function( __FUNCTION__, '3.11' );
```
Comment (borkweb) [inline src/Tribe/Query.php]: What is the replacement for this method?  Perhaps:

```
_deprecated_function( __METHOD__, '3.11', [some alternative] );
```

File: src/Tribe/Template/Month.php [hunk 2]
```diff
@@ -487,24 +551,24 @@ public static function day_classes() {
...
 			list ( $year, $month, $day ) = explode( '-', $calendar_day['date'] );
-			if ( self::$current_month == $month && self::$current_year == $year ) {
+			if ( $current_month == $month && $current_year == $year ) {
 				// Past, Present, Future class
-				if ( self::$today == $day ) {
+				if ( $today == $day ) {
 					$ppf .= ' tribe-events-present';
 				} else {
-					if ( self::$today > $day ) {
+					if ( $today > $day ) {
```
Comment (borkweb) [inline src/Tribe/Template/Month.php]: These `if`/`else` statements can be compressed into `ifelse` blocks if you wanted (not required).

File: src/Tribe/Template/Month.php [hunk 3]
```diff
@@ -331,24 +386,20 @@ protected function requested_date() {
...
 			}
 		}
 
-		protected function setup_tribe_bar_args() {
-			if ( empty( self::$tribe_bar_args ) ) {
-				foreach ( $_REQUEST as $key => $value ) {
-					if ( $value && strpos( $key, 'tribe' ) === 0 && $key != 'tribe-bar-date' ) {
-						self::$tribe_bar_args[ $key ] = $value;
-					}
+		protected function get_tribe_bar_args() {
```
Comment (borkweb) [inline src/Tribe/Template/Month.php]: Toss in a docblock comment while you're here tweaking this method?

File: src/Tribe/Template/Month.php [hunk 4]
```diff
@@ -71,8 +150,19 @@ public function __construct( $args = null ) {
...
-			self::$args                 = $args;
-			self::$posts_per_page_limit = apply_filters( 'tribe_events_month_day_limit', tribe_get_option( 'monthEventAmount', '3' ) );
+			self::$args            = $args;
+			$this->events_per_day  = apply_filters( 'tribe_events_month_day_limit', tribe_get_option( 'monthEventAmount', '3' ) );
+			$this->requested_date  = $this->requested_date();
+			$this->first_grid_date = $this->calculate_first_cell_date( $this->requested_date );
+			$this->final_grid_date = $this->calculate_final_cell_date( $this->requested_date );
+
+			// get all the ids for the events in this month, speeds up queries
+			$this->events_in_month = tribe_get_events( array_merge( $args, array(
```
Comment (borkweb) [inline src/Tribe/Template/Month.php]: Awesome change!

**Comment by borkweb** [comment]
@jazbek: this looks really great! Much more readable, too.  Fix the merge conflicts and twiddle a few things, then get rid of the `enhance` label on this PR. I'll try to keep an eye on when all that happens, but if I miss it, feel free to ping me.

File: src/Tribe/Query.php (hunk 1 above)
Comment (jazbek) [inline src/Tribe/Query.php]: There is actually no replacement for this function right now.. it returns an array of days of the month and the IDs & number of events on each day. Currently not doing that anywhere anymore.

File: src/Tribe/Template/Month.php (hunk 2 above)
Comment (jazbek) [inline src/Tribe/Template/Month.php]: Totally, I agree it's cleaner. Also kind of rewrote that function to improve readability / reduce nesting.

File: src/Tribe/Template/Month.php (hunk 3 above)
Comment (jazbek) [inline src/Tribe/Template/Month.php]: Done, also cleaned up docblock problems thru the rest of the file.

File: src/Tribe/Template/Month.php (hunk 4 above)
Comment (jazbek) [inline src/Tribe/Template/Month.php]: Thanks!

File: src/Tribe/Query.php (hunk 1 above)
Comment (jazbek) [inline src/Tribe/Query.php]: Just noticed __METHOD__ in your comment, good call, will fix.

**Comment by jazbek** [comment]
@borkweb leaving the code review tag on because I rewrote the day_classes() method. I double checked the frontend to make sure all the correct classes are being applied to the month days, but would feel safer with a 2nd set of eyes. Also wasn't able to provide an alternative method for the deprecated Tribe__Events__Query::getEventCounts(), so just wanted to make sure that was cool before merging.

**Comment by borkweb** [comment]
The new `day_classes()` method looks much nicer! Yeah, if there isn't a replacement method for `getEventCounts()`, that's fine - I just was double checking that the replacement wasn't omitted by accident.  Looking good!

**Comment by borkweb** [comment]
I've fixed the merge conflicts and have cherry-picked your work into a new branch so that it can be merged into `release/118`. Here's the PR: https://github.com/moderntribe/the-events-calendar/pull/156

**Pull request closed without merging**